### PR TITLE
Enhance katib client by adding get_success_trial_details()

### PR DESCRIPTION
**Pull request by Adarsh2910**
Added API to get list of trials that succeded with hyperparam and metrics to KatibClient

```
get_success_trial_details(self, name, namespace) -> List of trial name with hyperparams and metrics
```

### CODE REVIEW

**Comment by Adarsh2910** [comment]
@googlebot I signed it!

**Review by gaocegege — COMMENTED** [review]
/lgtm

/cc @andreyvelich

File: sdk/python/v1beta1/kubeflow/katib/api/katib_client.py [hunk 1]
```diff
@@ -329,6 +329,52 @@ def list_trials(self, name=None, namespace=None):
           {2} ".format(name, namespace, e))
         return result
 
+    def get_success_trial_details(self, name=None, namespace=None):
```
Comment (andreyvelich) [inline sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:332]: Maybe we should have an API which returns all Experiment's Trials objects instead of only Succeeded Trials with name, hyperparams, metrics?

Then user will have more freedom with this list of Trials ?
E.g. Get all Succeeded Trials, Get Early Stopped Trials, Get top <k> Trials.

WDYT @Adarsh2910 @gaocegege ?

**Review by andreyvelich — COMMENTED** [review]
Thank you for this implementation @Adarsh2910!
I left one comment.

File: sdk/python/v1beta1/kubeflow/katib/api/katib_client.py (hunk 1 above)
Comment (gaocegege) [reply sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:332]: SGTM, should we add it in this PR? or open an issue for it?

Comment (Adarsh2910) [reply sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:332]: Do you think it's a good idea to give the user a way to filter the trials based on status, or just return the list of trials with hyperparams and metrics?

If it's the latter, we can also modify the `list_trials` method and just add hyperparams and metrics to the return.

Comment (andreyvelich) [reply sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:332]: From my point of view, `list_trials` should return the list of `Trial` objects, instead of only name and status.
We can also add function: `get_success_trial_details` if it is needed in addition to `list_trials`.
What do you think @Adarsh2910 @gaocegege  ?

Comment (Adarsh2910) [reply sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:332]: That sounds good to me
I think having a function to just get the success trials will be a helpful addition and we can raise another issue for patching `list_trials` to get the list of `Trial` objects

Comment (Adarsh2910) [reply sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:332]: If we are all in agreement, Can we go ahead and merge this one in?
Thoughts? @andreyvelich @gaocegege

Comment (andreyvelich) [reply sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:332]: Make sense, we can update `list_trials` in the next PR.
WDYT @gaocegege ?

File: sdk/python/v1beta1/kubeflow/katib/api/katib_client.py [hunk 2]
```diff
@@ -329,6 +329,52 @@ def list_trials(self, name=None, namespace=None):
...
+        try:
+            katibtrial = thread.get(constants.APISERVER_TIMEOUT)
+            result = []
+            for i in katibtrial.get("items"):
+                output = {}
+                if i.get("metadata", {}).get("ownerReferences")[0].get("name") == name:
+                    status = i.get("status", {}).get("conditions", [])[-1].get("type")
+                    if status == "Succeeded":
+                        output["name"] = i.get("metadata", {}).get("name")
+                        output["hyperparams"] = i.get("spec", {}).get("parameterAssignments", [])
```
Comment (andreyvelich) [inline sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:359]: Maybe use here not short word, what do you think @Adarsh2910  ?
```suggestion
                        output["hyperparameters"] = i.get("spec", {}).get("parameterAssignments", [])
```

File: sdk/python/v1beta1/kubeflow/katib/api/katib_client.py [hunk 3]
```diff
@@ -329,6 +329,52 @@ def list_trials(self, name=None, namespace=None):
           {2} ".format(name, namespace, e))
         return result
 
+    def get_success_trial_details(self, name=None, namespace=None):
+        """
+        Get trial details that succeeded for an experiment
+        :param name: existing experiment name
+        :param namespace: defaults to current or default namespace
+        :return: trials name with hyperparams and metrics
```
Comment (andreyvelich) [inline sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:337]:
```suggestion
        :return: trials name with hyperparameters and metrics
```

File: sdk/python/v1beta1/kubeflow/katib/api/katib_client.py (hunk 1 above)
Comment (gaocegege) [reply sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:332]: SGTM.

File: sdk/python/v1beta1/kubeflow/katib/api/katib_client.py (hunk 2 above)
Comment (Adarsh2910) [reply sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:359]: yep, will fix it

File: sdk/python/v1beta1/kubeflow/katib/api/katib_client.py [hunk 4]
```diff
@@ -329,6 +329,52 @@ def list_trials(self, name=None, namespace=None):
...
+            async_req=True)
+
+        katibtrial = None
+        try:
+            katibtrial = thread.get(constants.APISERVER_TIMEOUT)
+            result = []
+            for i in katibtrial.get("items"):
+                output = {}
+                if i.get("metadata", {}).get("ownerReferences")[0].get("name") == name:                    
+                    output["status"] = i.get("status", {}).get(
```
Comment (andreyvelich) [inline sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:356]: Why did you delete check with succeeded status ?
Do you want to return hyperparameters and metrics for all Trials ?

Comment (Adarsh2910) [reply sdk/python/v1beta1/kubeflow/katib/api/katib_client.py]: ah, my bad
pushed a fix for it

**Review by andreyvelich — APPROVED** [review]
Thank you @Adarsh2910!
/lgtm
/approve
